conduit check takes 90 seconds to detect a missing K8s cluster

On Conduit 0.3.0, with minikube stopped, I get this behavior:

```
william:~$ time conduit check
kubectl: is in $PATH............................................................[ok]
kubectl: has compatible version.................................................[ok]
kubectl: can talk to Kubernetes cluster.........................................[FAIL]  -- Unable to connect to the server: dial tcp 192.168.99.100:8443: i/o timeout

kubernetes-api: can initialize the client.......................................[ok]
kubernetes-api: can query the Kubernetes API....................................[ERROR] -- HTTP GET request to endpoint [https://192.168.99.100:8443/api/v1/] resulted in error: [Get https://192.168.99.100:8443/api/v1/: dial tcp 192.168.99.100:8443: i/o timeout]

conduit-api: can retrieve status via gRPC.......................................[ERROR] -- Post https://192.168.99.100:8443/api/v1/namespaces/conduit/services/http:api:http/proxy/api/v1/SelfCheck: dial tcp 192.168.99.100:8443: i/o timeout

Status check results are [ERROR]

real	1m30.386s
user	0m0.351s
sys	0m0.063s
```

`check` should detect this in <90 seconds

--- a/pkg/healthcheck/grpc.go
+++ b/pkg/healthcheck/grpc.go
@@ -3,6 +3,7 @@ package healthcheck
 import (
 	"context"
 	"fmt"
+	"time"
 
 	healthcheckPb "github.com/runconduit/conduit/controller/gen/common/healthcheck"
 	"google.golang.org/grpc"
@@ -24,7 +25,10 @@ func (proxy *statusCheckerProxy) SelfCheck() []*healthcheckPb.CheckResult {
 		CheckDescription: "can query the Conduit API",
 	}
 
-	selfCheckResponse, err := proxy.delegate.SelfCheck(context.Background(), &healthcheckPb.SelfCheckRequest{})
+	ctx, cancel := context.WithTimeout(context.Background(), 5*time.Second)
+	defer cancel()
+
+	selfCheckResponse, err := proxy.delegate.SelfCheck(ctx, &healthcheckPb.SelfCheckRequest{})
 	if err != nil {
 		canConnectViaGrpcCheck.Status = healthcheckPb.CheckStatus_ERROR
 		canConnectViaGrpcCheck.FriendlyMessageToUser = err.Error()

--- a/pkg/k8s/api.go
+++ b/pkg/k8s/api.go
@@ -1,11 +1,13 @@
 package k8s
 
 import (
+	"context"
 	"encoding/json"
 	"fmt"
 	"io/ioutil"
 	"net/http"
 	"net/url"
+	"time"
 
 	healthcheckPb "github.com/runconduit/conduit/controller/gen/common/healthcheck"
 	"github.com/runconduit/conduit/pkg/healthcheck"
@@ -93,7 +95,11 @@ func (kubeapi *kubernetesApi) checkApiAccess(client *http.Client) (*healthcheckP
 		return checkResult, ""
 	}
 
-	resp, err := client.Get(endpointToCheck.String())
+	req, _ := http.NewRequest("GET", endpointToCheck.String(), nil)
+	ctx, cancel := context.WithTimeout(context.Background(), 5*time.Second)
+	defer cancel()
+
+	resp, err := client.Do(req.WithContext(ctx))
 	if err != nil {
 		checkResult.Status = healthcheckPb.CheckStatus_ERROR
 		checkResult.FriendlyMessageToUser = fmt.Sprintf("HTTP GET request to endpoint [%s] resulted in error: [%s]", endpointToCheck, err.Error())

--- a/pkg/version/version.go
+++ b/pkg/version/version.go
@@ -90,7 +90,10 @@ func (v versionStatusChecker) SelfCheck() []*healthcheckPb.CheckResult {
 }
 
 func (v versionStatusChecker) getServerVersion() (string, error) {
-	resp, err := v.publicApiClient.Version(context.Background(), &pb.Empty{})
+	ctx, cancel := context.WithTimeout(context.Background(), 5*time.Second)
+	defer cancel()
+
+	resp, err := v.publicApiClient.Version(ctx, &pb.Empty{})
 	if err != nil {
 		return "", err
 	}

